django/forms/models.py

```diff
@@ -279,6 +279,13 @@ def __new__(mcs, name, bases, attrs):
                 # fields from the model"
                 opts.fields = None
 
+            if opts.exclude:
```

Comment:
Confirmed working, and I like your way better because it keeps the logic at the same place. However, we should also have tests for the similar case where an overridden field is not in Meta.fields.